tasks/managed/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -297,21 +305,236 @@ spec:
+                echo "INFO: Launching worker $launched_count/$total_groups for OCP group: $ocp_version"
 
                 local group_components
                 group_components=$(cat "$group_file")
 
-                process_ocp_group "$ocp_version" "$group_components"
+                (
+                    set -e
+                    
+                    echo "INFO: [Parallel Worker $ocp_version] Starting processing"
+                    
+                    process_ocp_group "$ocp_version" "$group_components"
+                    
+                    echo "SUCCESS" > "${parallel_dir}/${ocp_version}.status"
+                    echo "INFO: [Parallel Worker $ocp_version] Completed successfully"
+                ) &
+                
+                local pid=$!
+                pids+=("$pid")
+                pid_to_ocp+=("$pid:$ocp_version")
+                echo "INFO: Launched worker for OCP $ocp_version (PID: $pid)" \
+                     "[Active workers: ${#pids[@]}/$max_parallel_workers]"
             done
 
-            echo "INFO: Processing completed successfully"
+            echo "INFO: All $total_groups workers launched, waiting for final ${#pids[@]} to complete..."
+
+            # Wait for remaining background processes and track failures
+            for pid in "${pids[@]}"; do
+                local ocp_version="unknown"
+                for mapping in "${pid_to_ocp[@]}"; do
+                    if [[ "$mapping" == "$pid:"* ]]; then
+                        ocp_version="${mapping#*:}"
+                        break
+                    fi
+                done
+                
+                if wait "$pid"; then
+                    echo "INFO: Worker for OCP $ocp_version (PID: $pid) completed successfully"
+                    completed_count=$((completed_count + 1))
+                else
+                    local exit_code=$?
+                    echo "ERROR: Worker for OCP $ocp_version (PID: $pid) failed with exit code $exit_code"
+                    failed_groups+=("$ocp_version")
+                fi
+            done
+
+            # Check if any groups failed
+            # ToDo: verify if we can proceed with partial release
+            if [ ${#failed_groups[@]} -gt 0 ]; then
+                echo "ERROR: ${#failed_groups[@]} OCP group(s) failed processing" \
+                     "(adaptive limit: $max_parallel_workers):"
+                for failed_group in "${failed_groups[@]}"; do
+                    echo "  - OCP version: $failed_group"
+                done
+                echo "Cannot proceed with partial release - all OCP groups must succeed"
+                exit 1
+            fi
+
+            echo "INFO: All parallel processing completed successfully for $total_groups OCP groups"
+            echo "INFO: Adaptive parallelism used $max_parallel_workers concurrent workers"
+
+            echo "INFO: Merging results from all parallel workers"
+            for ocp_version in $ocp_versions; do
```

Comment:
I would expect a shellcheck issue here. In any case, why not use the bash array here as well?

Reply:
Not exactly: a shellcheck will not complain, since word splitting in for loops is often expected behavior.
But you are absolutely right, we should use the array for consistency and better practice. 
fixed, thank you!